ucp/core.py

```diff
@@ -674,6 +674,60 @@ def close_after_n_recv(self, n, count_from_ep_creation=False):
                 % (n, self._finished_recv_count)
             )
 
+    async def send_obj(self, obj, tag=None):
+        """Send `obj` to connected peer that calls `recv_obj()`.
+
+        The transfer includes an extra message containing the size of `obj`,
+        which increses the overhead slightly.
+
+        Parameters
+        ----------
+        obj: exposing the buffer protocol or array/cuda interface
+            The object to send.
+        tag: hashable, optional
+            Set a tag that the receiver must match.
+
+        Example
+        -------
+        >>> await ep.send_obj(pickle.dumps([1,2,3]))
+        """
+
+        nbytes = get_buffer_nbytes(
+            buffer=obj, check_min_size=None, cuda_support=self._cuda_support
+        )
+        await self.send(struct.pack("Q", nbytes), tag=tag)
+        await self.send(obj, tag=tag)
+
+    async def recv_obj(self, tag=None, allocator=bytearray):
+        """Receive from connected peer that calls `send_obj()`.
+
+        As opposed to `recv()`, this function returns the received object.
+
+        The transfer includes an extra message containing the size of `obj`,
+        which increses the overhead slightly.
+
+        Parameters
+        ----------
+        tag: hashable, optional
+            Set a tag that must match the received message. Notice, currently
+            UCX-Py doesn't support a "any tag" thus `tag=None` only matches a
+            send that also sets `tag=None`.
```

Comment:
If `tag=None`, would that just hash `None` and use it as tag? If so, isn't that dangerous due to collisions? I think we had deadlocks due to collisions in the past.

Reply:
No,  `tag=None` is no tags but since UCX-Py uses tags to pair endpoint connections we don't support an _any tag_.